firmware/src/src/sense.cpp

```diff
@@ -433,13 +433,33 @@ void calculate_Thread()
         int tltch = trkset.tiltCh();
         int rllch = trkset.rollCh();
         int panch = trkset.panCh();
+        int alertch = trkset.alertCh();
         if(tltch > 0)
             channel_data[tltch - 1] = trpOutputEnabled == true ? tiltout_ui : trkset.Tlt_cnt();
         if(rllch > 0)
             channel_data[rllch - 1] = trpOutputEnabled == true ? rollout_ui : trkset.Rll_cnt();
         if(panch > 0)
             channel_data[panch - 1] = trpOutputEnabled == true ? panout_ui : trkset.Pan_cnt();
 
+        // TODO what is trpOutputEnabled?
+        static float pulsetimer=0;
+        static bool sendingresetpulse = false;
+        if (alertch > 0) {
+            // Synthesize a pulse indicating recenter started
+            channel_data[alertch - 1] = TrackerSettings::MIN_PWM;
+            if (butdnw) {
+                sendingresetpulse = true;
+                pulsetimer=0;
+            }
+            if (sendingresetpulse) {
+                channel_data[alertch - 1] = TrackerSettings::MAX_PWM;
+                pulsetimer += (float)CALCULATE_PERIOD / 1000000.0;
+                if(pulsetimer > TrackerSettings::RECENTER_PULSE_DURATION) {
+                    sendingresetpulse = false;
+                }
+            }
+        }
+
```

Comment:
Haven't actually tested but this looks good to me! Although I think it should be moved above tilt / roll / pan setting line 433, I know the GUI/FW shouldn't ever let you pick two channels the same but I believe TRP should always superseded any possible channels set the same, should it occur.